src/NJsonSchema/Infrastructure/XmlDocumentationExtensions.cs

```diff
@@ -562,6 +562,11 @@ private static async Task<string> GetXmlDocumentationPathAsync(dynamic assembly)
             }
             catch
             {
+                var currentDirectory = await DynamicApis.DirectoryGetCurrentDirectoryAsync();
+                path = DynamicApis.PathCombine(currentDirectory, "bin\\" + assembly.GetName().Name + ".xml");
```

Comment:
This won't always work. The application and its assemblies are not always located in the bin folder. It can be anything. The only good fallback for a case when assembly Location and CodeBase are unavailable is current app domain's BaseDirectory. 

I'd suggest suppressing exceptions while trying to access assembly location and code base and always try to obtain app domain base directory, but also in a try/catch block, like this:

``` csharp
try
{
    // try to get assembly location
    // or try to get assembly code base
}
catch
{
    // ignore exceptions, don't return yet
}

try
{
    // try to get app domain base directory
}
catch
{
    // everything failed
    return null;
}
```